theano/gpuarray/extra_ops.py

```diff
@@ -456,6 +454,8 @@ def c_support_code_struct(self, node, nodename):
 @register_opt2([CumsumOp], 'fast_compile')
 def use_gpu_cumsumop(op, ctx_name, inputs, outputs):
     if inputs[0].dtype == 'float32':
+        if isinstance(inputs[0].type, GpuArrayType):
+            return
```

Comment:
Why is that needed?
It looks like it prevents the optimization from being applied in the new optimizer.
Since it would still be part of the old one, the tests would pass, but the point is to have the local optimizations applied in the new optimizer.